src/scval_validations.rs

```diff
@@ -63,39 +63,134 @@ impl Validate for ScVal {
 }
 
 impl Validate for ScMap {
-    type Error = ();
+    type Error = Error;
 
     fn validate(&self) -> Result<(), Self::Error> {
-        // TODO: Validate that the map is sorted and has no duplicates, or find
-        // a way to guarantee this to be the case.
-        todo!()
+        // Check the map is sorted by key, and there are no keys that are
+        // duplicates.
+        if self.windows(2).all(|w| w[0].key < w[1].key) {
```

Comment:
What is this `.windows` function? Can't find its definition.

Reply:
The function is https://doc.rust-lang.org/std/primitive.slice.html#method.windows.

Reply:
Ah cool!